### PR TITLE
Improve IntSet and IntMap module docs.

### PR DESCRIPTION
**Pull request by m-renaud**
Rendered docs can be found at [Data.IntSet](http://www.m-renaud.com/haddocks/containers-0.5.11.0/Data-IntSet.html), [Data.IntMap.Strict](http://www.m-renaud.com/haddocks/containers-0.5.11.0/Data-IntMap-Strict.html), [Data.IntMap.Lazy](http://www.m-renaud.com/haddocks/containers-0.5.11.0/Data-IntMap-Lazy.html).

### CODE REVIEW

**Comment by m-renaud** [comment]
Any concerns @treeowl or can I go ahead and merge this in?

**Comment by treeowl** [comment]
I haven't looked yet.

File: Data/IntMap/Lazy.hs [hunk 1]
```diff
@@ -14,26 +14,47 @@
...
--- If you need value-strict maps, use "Data.IntMap.Strict" instead.
--- The 'IntMap' type itself is shared between the lazy and strict modules,
--- meaning that the same 'IntMap' value can be passed to functions in
--- both modules (although that is rarely needed).
+-- = Finite Int Maps (lazy interface)
 --
--- These modules are intended to be imported qualified, to avoid name
--- clashes with Prelude functions, e.g.
+-- The @'IntMap' v@ type represents a finite map (sometimes called a dictionary)
+-- from key of type @Int@ to values of type @v@.
```
Comment (treeowl) [inline Data/IntMap/Lazy.hs]: s/from key/from keys

File: Data/IntMap/Strict.hs [hunk 2]
```diff
@@ -15,26 +15,63 @@
...
+-- Benchmarks comparing "Data.IntMap.Strict" with other dictionary
+-- implementations can be found at https://github.com/haskell-perf/dictionaries.
+--
+--
+-- == Warning
+--
+-- The 'IntMap' type is shared between the lazy and strict modules, meaning that
+-- the same 'IntMap' value can be passed to functions in both modules. This
+-- means that the 'Functor', 'Traversable' and 'Data' instances are the same as
+-- for the "Data.IntMap.Lazy" module, so if they are used on strict maps, the
```
Comment (treeowl) [inline Data/IntMap/Strict.hs]: There are no "strict maps".

File: Data/IntMap/Lazy.hs (hunk 1 above)
Comment (m-renaud) [reply Data/IntMap/Lazy.hs]: Done.

File: Data/IntMap/Strict.hs (hunk 2 above)
Comment (m-renaud) [reply Data/IntMap/Strict.hs]: Fixed.

**Pull request merged**